src/oscar/apps/voucher/abstract_models.py

```diff
@@ -119,7 +89,7 @@ class AbstractVoucher(models.Model):
     Oscar enforces those modes by creating VoucherApplication
     instances when a voucher is used for an order.
     """
-    name = models.CharField(_("Name"), max_length=128,
+    name = models.CharField(_("Name"), max_length=128, unique=True,
```

Comment:
The `Voucher` model currently has a `benefit` property which returns the benefit for the first offer - on the assumption that a voucher will likely contain only one offer. I think we should deprecate that property since that assumption is no longer valid (i.e., apply the `oscar.core.decorators.deprecated` decorator to it.